dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/datastatistics/hibernate/HibernateDataStatisticsStore.java

```diff
@@ -99,6 +99,16 @@ else if ( eventInterval == EventInterval.MONTH )
             ads.setMapViews( resultSet.getInt( "mapViews" ) );
             ads.setChartViews( resultSet.getInt( "chartViews" ) );
             ads.setPivotTableViews( resultSet.getInt( "reportTableViews" ) );
+
+            // Logic needed to assist the deprecation process of charts and
+            // report tables.
+            int visualizationViews = resultSet.getInt( "visualizationViews" );
+            if ( visualizationViews == 0 )
+            {
+                visualizationViews = resultSet.getInt( "chartViews" ) + resultSet.getInt( "reportTableViews" );
```

Comment:
Could it maybe be better to make a Flyway db routine to migrate this once and forever to visualization, and keep the code cleaner and faster?

Reply:
Good suggestion Lars, I changes were applied.